sdk/go/pkg/neo4j/neo4j.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2024 Hypermode Inc.
+ * Licensed under the terms of the Apache License, Version 2.0
+ * See the LICENSE file that accompanied this code for further details.
+ *
+ * SPDX-FileCopyrightText: 2024 Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package neo4j
+
+import (
+	"fmt"
+
+	"github.com/hypermodeinc/modus/sdk/go/pkg/utils"
+)
+
+type Neo4jOption func(*neo4jOptions)
+
+type neo4jOptions struct {
+	dbName string
+}
+
+func WithDbName(dbName string) Neo4jOption {
+	return func(o *neo4jOptions) {
+		o.dbName = dbName
+	}
+}
+
+type EagerResult struct {
+	Keys    []string
+	Records []*Record
+}
+
+type Record struct {
+	Values []string
```

Comment:
How can all record values be strings?

Reply:
We discussed this, the records are of type any but to cross the wasm boundary it has to be typed, so we made it type string